### ISSUE TITLE
Question : Yolo as a pure image classifier

### ISSUE THREAD

**Issue by holger-prause** [AlexeyAB/darknet#4703]
I need to do a pure image classification(telling if an image is of certain class or not) and i dont care about bounding boxes at all(actually the dataset does not contain any - only a label for the whole image)

Can i archieve this with yolo by just putting one big bounding box over the whole image with the desired class? Or will this mess up the training? Thank you very much - i am still learning.
 
Greetings,
Holger

**Comment by kossolax** [comment]
Just use the classifier part: https://pjreddie.com/darknet/imagenet/

**Comment by holger-prause** [comment]
Ok i think i start to understand. 
I can drop the object detection party completly. 
If i take a look at the config for example this one - https://github.com/pjreddie/darknet/blob/master/cfg/alexnet.cfg i dont see any yolo layers, this verifies it for me.

The linked page describes how to do prediction on a pretrained model. I need to train such a model by myself. I think i should just use the same config(with adoption to my amount of classes) for training as used for inference.

Things which are still unclear to me:
- What is the input format to train such a classifier (in contrast to the input format used for object detection , (centerX, centerY,relw,relh)
- How to prepare the weights for this - i dont want to start from scratch.
  I think i should use the partial command of the corresponding weights but i am not sure which layers to drop. If i remember right - in general you should drop softmax + the fully connected layer on the end.

Am i on the right track here?
Thank you very much + greetings,
Holger

**Comment by kossolax** [comment]
For classifier, groundtrusth is directly in file name.

Le jeu. 16 janv. 2020 à 13:05, Holger Prause <notifications@github.com> a
écrit :

**Comment by holger-prause** [comment]
About the input format - it should be the index of the label - but i dont know how to give that to the network without coding c++(which i am not so good at).

**Comment by kossolax** [comment]
xxxxxx_dog.jpg

Le jeu. 16 janv. 2020 à 13:28, Holger Prause <notifications@github.com> a
écrit :

**Comment by holger-prause** [comment]
Got it - fair enough. Thx man.
So the format is [basename]_[classname].[ext].
In the case of multiclass - i just need to make a copy of the image and rename it to the 2nd class?
Or can i include mutiple classes in the filename?

If you could comments on the weights section too("how to prepare weights") - this would be great and i would have everything to go. But you already helped me a lot so i dont want to be greedy.

**Comment by kossolax** [comment]
Darknet classifier doesn't support well classifier: https://github.com/AlexeyAB/darknet/pull/3728

**Comment by holger-prause** [comment]
Ok - thats unfortunate but acceptable for me right now. About the weights i will try to find about by myself or raise another question later.

Thank you very much - closing this issue :-)
Have a nice day!
